solutions/devsprint-matheus-reis-7/FinanceApp/Modules/ActivityDetails/ActivityDetailsViewController.swift

```diff
@@ -6,19 +6,77 @@
 //
 
 import UIKit
+import RxCocoa
+import RxSwift
 
-class ActivityDetailsViewController: UIViewController {
+final class ActivityDetailsViewController: UIViewController {
 
-    lazy var activityDetailsView: ActivityDetailsView = {
+    // MARK: - UIView properties
+    private let activityDetailsView: ActivityDetailsViewProtocol
 
-        let activityDetailsView = ActivityDetailsView()
-        activityDetailsView.delegate = self
-        return activityDetailsView
-    }()
+    // MARK: - DataSource / UseCase dependencies
+    private let service: FinanceService
+
+    // MARK: - Reactive Properties
+    private let disposeBag: DisposeBag
+    private let activityDetailsObservable = BehaviorRelay<ActivityDetails>(
+        value: .init(name: "",
+                     price: 0,
+                     category: "",
+                     time: "")
+    )
+
+    // MARK: - Initializers
+    // TODO: Abstract FinanceService as ServiceProtocol
+    init(activityDetailsView: ActivityDetailsViewProtocol = ActivityDetailsView(),
+         service: FinanceService = FinanceService(),
+         disposeBag: DisposeBag = DisposeBag()) {
+        self.activityDetailsView = activityDetailsView
+        self.service = service
+        self.disposeBag = disposeBag
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) { nil }
 
+    // MARK: - UIViewController lifecycle methods
     override func loadView() {
         self.view = activityDetailsView
     }
+
+    override func viewDidLoad() {
+        activityDetailsView.delegate = self
+        fetchDataView()
+        bindObservables()
+    }
+}
+
+// MARK: -
+private extension ActivityDetailsViewController {
+
+    private func fetchDataView() {
+        service.fetchActivityDetails().subscribe(onSuccess: { [weak self ] activityDetails in
+            guard let self = self else { return }
+            self.activityDetailsObservable.accept(activityDetails)
+        }, onFailure: { failure in
+            // TODO: What should be the approach on errors?
+            print("failure:", failure.localizedDescription)
```

Comment:
I would recommend to use the same approach you used for `activityDetails` so far

Reply:
Ok. In this case can I create an `acitivityDetails: ActivityDetails` instance with empty strings? Is a suitable solution?

```swift
let errorActivityDetails: ActivityDetails = .init(
    name: "Error",
    price: 0.0,
    category: "Error",
    time: "Error"
)   

// ...
self.activityDetailsObservable.accept(errorActivityDetails)
```

Did I get it right, @arctouch-gustavosoares?

Reply:
I don't think it would be a suitable solution... ActivityDetails it's a model to handle an entity right? I feel it should not be involved as an error. Because you're treating both cases in one structure, you know? It's better to create a separate Model for it. For example an Error entity, and create a bind to just show it. What do you think?

Reply:
Got it. I'll figure it out.